pkg/base/license.go

```diff
@@ -11,18 +11,32 @@
 package base
 
 import (
+	"context"
+	"time"
+
 	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
 	"github.com/cockroachdb/cockroach/pkg/util/uuid"
 	"github.com/cockroachdb/errors"
 )
 
+var ErrIsOSS = errors.New("OSS binaries do not include enterprise features")
+
 // CheckEnterpriseEnabled returns a non-nil error if the requested enterprise
 // feature is not enabled, including information or a link explaining how to
 // enable it.
 //
 // This function is overridden by an init hook in CCL builds.
 var CheckEnterpriseEnabled = func(_ *cluster.Settings, _ uuid.UUID, org, feature string) error {
-	return errors.New("OSS binaries do not include enterprise features")
+	return ErrIsOSS
+}
+
+// TimeToLicenseExpiry returns a duration object that measures the time until
+// the currently set enterprise license expires starting from the 3rd argument
+// passed in.
+//
+// This function is overridden by an init hook in CCL builds
+var TimeToLicenseExpiry = func(ctx context.Context, _ *cluster.Settings, _ time.Time) (time.Duration, error) {
+	return 0, ErrIsOSS
```

Comment:
I'd just return `0, nil` here? it isn't licensed -- so it should show zero time remaining just like if the setting is missing, but it doesn't seem like an error ?

Reply:
ah, so you want to still print the metric for OSS clusters, just always showing zero?

Reply:
Updated to return 0 in OSS and no license cases.